File: db4on/core/src/core/com/db4o/internal/cs/messages/MObjectSetGetId.cs
namespace com.db4o.@internal.cs.messages
{
	/// <exclude></exclude>
	public class MObjectSetGetId : com.db4o.@internal.cs.messages.MObjectSet
	{
		public override bool ProcessAtServer(com.db4o.@internal.cs.ServerMessageDispatcher
			 serverThread)
		{
			com.db4o.@internal.query.result.AbstractQueryResult queryResult = QueryResult(serverThread
				, ReadInt());
			int id = queryResult.GetId(ReadInt());
			serverThread.Write(com.db4o.@internal.cs.messages.Msg.OBJECTSET_GET_ID.GetWriterForInt
				(Transaction(), id));
			return true;
		}
	}
}

File: db4o.cs/src/com/db4o/internal/cs/messages/MObjectSetGetId.java
/* Copyright (C) 2006  Versant Inc.  http://www.db4o.com */

package com.db4o.internal.cs.messages;

import com.db4o.internal.query.result.*;


/**
 * @exclude
 */
public class MObjectSetGetId extends MObjectSet implements MessageWithResponse {
	
	public boolean processAtServer() {
		AbstractQueryResult queryResult = queryResult(readInt());		
		int id = 0;
		synchronized (streamLock()) {
			id = queryResult.getId(readInt());
		}
		write(Msg.OBJECTSET_GET_ID.getWriterForInt(transaction(), id));
		return true;
	}

}
